fetch.bs

```diff
@@ -6680,6 +6720,308 @@ agent's <a>CORS-preflight cache</a> for which there is a <a>cache entry match</a
+   <var>relative</var> which is a <a>navigable container</a>.
+
+   <li><p><a for=list>For each</a> <var>descendantContainer</var> of <var>relevantContainers</var>,
+   decrement <var>remainingTotalQuota</var> by the <a>initial subframe deferred fetch quota</a>
+   given <var>descendantContainer</var>'s <a>deferred fetch policy</a>.
+  </ol>
+ </li>
+
+ <li><p>If <var>remainingTotalQuota</var> is less than <var>remainingQuotaForOrigin</var>, then
+ return <var>remainingTotalQuota</var>.
+
+ <li><p>Otherwise, return <var>remainingQuotaForOrigin</var>.
+</ol>
+</div>
+
+<div algorithm>
+<p>To <dfn export>determine subframe deferred fetch policy</dfn> for a <a>navigable container</a>
+<var>container</var> and an <a for=/>origin</a> <var>originToNavigateTo</var>:
+
+<p class=note>This is called whenever a <a>navigable container</a> is being navigated, e.g. by
+setting an iframe's <a attribute for=HTMLIFrameElement>src</a>.
+
+<ol>
+ <li><p>Set <var>container</var>'s <a>deferred fetch policy</a> to <code>disabled</code>.
+
+ <li><p>If the <a data-lt="define an inherited policy for feature in container">inherited policy</a>
+ for {{PermissionPolicy/"deferred-fetch"}}, <var>navigable</var>'s <a>navigable container</a> and
+ <var>originToNavigateTo</var> is <code>Enabled</code>, and the
+ <a>available deferred fetching quota</a> for <var>container</var>'s <a>container document</a> is
+ equal or greater than 64 kibibytes, then set <var>navigable</var>'s
+ <a>deferred fetch policy</a> to {{PermissionPolicy/"deferred-fetch"}} and return.
+
+ <li><p>If the <a data-lt="define an inherited policy for feature in container">inherited policy</a>
+ for {{PermissionPolicy/"deferred-fetch-minimal"}}, <var>container</var> and
+ <var>originToNavigateTo</var> is <code>Disabled</code>, then set <var>container</var>'s
+ <a>deferred fetch policy</a> to <code>disabled</code> and return.
+
+
+ <li><p>Let <var>topLevelRelatives</var> be <var>container</var>'s <a>container document</a>'s
+ <a>deferred fetch quota-sharing navigables</a>.
+
+ <li>
+  <p>If <var>topLevelRelatives</var> does not <a for=list>contain</a> <var>container</var>'s
+  <a>node navigable</a>'s <a for=navigable>top-level traversable</a>, then
+  set <var>container</var>'s <a>deferred fetch policy</a> to <code>disabled</code> and
+  return.
+
+  <p class=note>Only the top level document and related documents can delegate the
+  {{PermissionPolicy/"deferred-fetch-minimal"}} policy.
+
+ <li><p>Let <var>framesWithMinimalQuotaPolicy</var> be zero.
+ <li>
+  <p><a for=list>For each</a> <var>navigable</var> that matches the following conditions:
+  <ul class=brief>
+
+   <li><p><var>navigable</var> is not <var>container</var>'s <a>content navigable</a>.
+
+   <li><p><var>topLevelRelatives</var> <a for=list>contains</a> <var>navigable</var>'s
+   <a for=navigable>parent</a>
+
+   <li><p><var>topLevelRelatives</var> does not <a for=list>contain</a> <var>navigable</var>
```

Comment:
(Assume my question at [here](https://github.com/whatwg/fetch/pull/1647/files#r1863326213) is resolved first)

Given the following example:

```
<html>
  <iframe id="frame-1" src="frame-1.html"></iframe>
  <iframe id="frame-2" src="frame-2.html"></iframe>
  <iframe id="frame-3" src="https://cross-domain.com/frame-3.html"></iframe>
  <iframe id="frame-4" src="https://cross-domain.com/frame-4.html"></iframe>
</html>
```

I assume `framesWithMinimalQuotaPolicy` should be `1` when reaching `frame-4`, but the for loop does not give this result.

```
For each navigable that matches the following conditions:

- navigable is not container’s content navigable.
- topLevelRelatives contains navigable’s parent
- topLevelRelatives does not contain navigable
- navigable’s navigable container’s deferred fetch policy is "deferred-fetch-minimal"
```

- Is `topLevelRelatives` for `frame-3` or `frame-4` = `{frame-3, frame-4}`?
  - If so, `topLevelRelatives contains navigable’s parent` will never hold.

Reply:
Let's take `frame-3`:
- `topLevelRelatives` is `{main document, frame-1, frame-2}` (see previous comment).
- The for loop iterates over all the navigables, not just on the `topLevelRelatives`. So it iterates over `{main document, frame-1, frame-2, frame-3, frame-4}`
- `frame-3` fails the first condition
- The main document passes the second condition
- `frame-1` and `frame_2` fail the 3rd condition
- We're left with `frame-4`, and we check its deferred fetch policy